assets/js/stripe-payment-request.js

```diff
@@ -585,10 +569,14 @@ jQuery( function( $ ) {
 
 		attachProductPageEventListeners: function( prButton, paymentRequest ) {
 			var paymentRequestError = [];
-			var addToCartButton = $( '.single_add_to_cart_button' );
 
 			prButton.on( 'click', function ( evt ) {
+				evt.preventDefault(); // Prevent showing payment request modal.
```

Comment:
Given that product information is retrieved asynchronously now. We need to stop the event and later display the payment dialog manually, only after getting the response of adding the product to the cart.